src/roswire/ros2/state.py

```diff
@@ -139,37 +154,57 @@ def probe(self) -> ROS2SystemState:
             for line in lines:
                 if "Publishers:" in line:
                     mode = _PUBLISHERS
+                    types = topic_to_type
                     continue
                 elif "Subscribers:" in line:
                     mode = _SUBSCRIBERS
+                    types = topic_to_type
                     continue
                 elif "Services:" in line:
                     mode = _SERVICES
+                    types = service_to_type
                     continue
                 elif "Action Servers:" in line:
                     mode = _ACTION_SERVERS
+                    types = action_to_type
                     continue
                 elif "Service Clients:" in line:
                     mode = _SERVICE_CLIENTS
+                    types = service_to_type
                     continue
                 elif "Action Clients:" in line:
                     mode = _ACTION_CLIENTS
+                    types = action_to_type
                     continue
 
                 if mode:
-                    name = line.partition(":")[0]
+                    partition = line.partition(":")
+                    name = partition[0]
+                    fmt = partition[1]
                     if name in node_to_state[mode]:
                         node_to_state[mode][name].add(node_name)
                     else:
                         node_to_state[mode][name] = {node_name}
+                    # Add type information and report conflict if a
+                    # different types was registered (probably
+                    # should never happen)
+                    if name in types and fmt != types[name]:
```

Comment:
This is a nice sanity check, but I think we should either raise an exception or use an assertion here. (Continuing with the execution could lead to strange results.)

Reply:
Chose to raise an exception.

Reply:
Raising an exception